### PR TITLE
ASTGen: account for alternate path separators

### PR DESCRIPTION
**Pull request by compnerd**
Windows supports `\` and `/` as path separators.  Account for both in computing the basename.  This repairs a few test failures on Windows.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift [hunk 1]
```diff
@@ -43,7 +43,7 @@ extension String {
   /// Retrieve the base name of a string that represents a path, removing the
   /// directory.
   var basename: String {
-    guard let lastSlash = lastIndex(of: "/") else {
+    guard let lastSlash = lastIndex(where: { ["/", "\\"].contains($0) }) else {
```
Comment (CodaFi) [inline lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift:46]: This condition should be specialized for Windows specifically

Comment (compnerd) [reply lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift:46]: Avoiding the special casing is nice - is there some case that you expect this to be problematic for on other platforms?

Comment (compnerd) [reply lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift]: Special cased Apple and Linux platforms instead.

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**